src/compiler/checker.ts

```diff
@@ -909,9 +933,33 @@ namespace ts {
                     }
                 }
                 if (target.aliasSymbol) {
-                    relevant.add(target.aliasSymbol)
+                    relevant.add(target.aliasSymbol);
+                    // If the current type is a union type, add the inherited type symbols common to all members
+                    if (target.flags & TypeFlags.Union) {
+                        const types = (target as UnionType).types;
+                        const inherited: Set<Symbol>[] = []
+                        for (const member of types) {
+                            if (!seen.has(member)) {
+                                seen.add(member);
+                                inherited.push(collectRelevantSymbolsLoop(member));

```

Comment:
This is not stack-safe, however, I think it would be difficult to reach the recursion depth limit here. `collectRelevantSymbolsLoop` can be rewritten to be completely iterative (I tried), but it gets really complicated. Imposing a strict limit on how many times this function can recurse, and showing a diagnostic if it hits the limit, could be another way to avoid unbounded recursion if that's a concern.